Collection mobile

daiagi opened the pull request:
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [x] Refactoring

## Context

- [x] Closes #5076 
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [x] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

![image](https://user-images.githubusercontent.com/22791238/222641454-0615576a-a091-456b-bf81-d87fb839c18d.png)
![image](https://user-images.githubusercontent.com/22791238/222641498-5b5fcdb6-786d-4c8b-b8cc-45c829850ce2.png)

daiagi commented:
will require merge resolution with #5146 
need to merge #5120  before merging this

exezbcz commented:
@daiagi yup, that is correct, there is no grid button on mobile - mobile nft cards are also bit different
<img width="353" alt="image" src="https://user-images.githubusercontent.com/90852205/222652601-12a8e44e-77e6-4155-9339-6a9c5b2f44a4.png">
I will have a look 👍

exezbcz commented:
oki, menu icon into a button - like on mobile fixes, stroke + dropshadow
<img width="533" alt="image" src="https://user-images.githubusercontent.com/90852205/222653143-5127c482-c6b5-4e60-8972-d8a5068040ff.png">

+ change the nft cards on mobile to the minimized version like on explore
<img width="413" alt="image" src="https://user-images.githubusercontent.com/90852205/222653594-79ffd8be-eaff-4f24-80a9-9c32872aef27.png">

daiagi commented:
[] count icon on filter menu toggle button
Like sort have

daiagi commented:
:heavy_check_mark:  oki, menu icon into a button - like on mobile fixes, stroke + dropshadow
:heavy_check_mark: change the nft cards on mobile to the minimized version like on explore
:heavy_check_mark: and with the animation, there is a like a flash in darkmode before the screen moves in in (solid bg with lowered opacity - white), not sure if this was applied to desktop as well but the icons in menu are delayed and then jump over the whole screen
:heavy_check_mark: count icon on filter menu toggle button (mobile only)
:heavy_check_mark: sort button after chain selector on explore (and collection) in mobile

exezbcz commented:
niceee, its looking good

- please change the menu so it looks same as on explore 
  - changer are - "filter" to bold, remove dropshadow around reset all button
  
![image](https://user-images.githubusercontent.com/90852205/222961613-c88d4289-7802-40ee-8b3d-e6846e72da3d.png)

- when is sorting applied and filters not, the menu icon is not aligned with content around 
![image](https://user-images.githubusercontent.com/90852205/222961683-adf99e7c-bea6-4268-bd68-3620de0c5834.png)
- looks like its broken on bigger touch screens as well
![image](https://user-images.githubusercontent.com/90852205/222961736-5beb7058-2e50-476f-a5b2-95f3597e43cc.png)

animation looks good now 👍 

Please remove the divider that was separating the breadcrumbs and content on desktop
![image](https://user-images.githubusercontent.com/90852205/222961830-83089c32-6437-4a49-a76c-0533d55e7dd3.png)

And then we may delete buy now by default from collections? Thinking out loud. Wdyt @roiLeo

daiagi commented:
> when is sorting applied and filters not, the menu icon is not aligned with content around

yep, notice it too, fixed already

exezbcz commented:
the item count now is on weird spot due to removing the divider on all touch devices I think.
![image](https://user-images.githubusercontent.com/90852205/222967044-a850dd76-bf95-4bc8-90b9-ee5b2ab15683.png)
- also on desktop the divider is missing now

![image](https://user-images.githubusercontent.com/90852205/222967114-cf58a645-a931-47df-9a51-9d4bd4687818.png)


+ the creator name changes to black bold on mobile, it should remain blue normal weight
![image](https://user-images.githubusercontent.com/90852205/222967169-9a55727d-6a2e-4e01-85a7-dee7bfff6af3.png)

daiagi commented:
It's due to cases where the identity pop over is present and cases where it's renderd as simple link

Is the blue color app wide? Can I change the default color of the link to be blue?

Thank you

exezbcz commented:
@daiagi ah, oki

the link isnt blue everywhere, see gallery item detail - creator and owner are bold black

daiagi commented:
> the item count now is on weird spot due to removing the divider on all touch devices I think.

apologies, it slipped my vision. fixed

>also on desktop the divider is missing now

sure? i could not reproduce it

* Firefox
![image](https://user-images.githubusercontent.com/22791238/223045410-e252e753-5489-418a-b60e-3c8b231056e4.png)


* Brave
![image](https://user-images.githubusercontent.com/22791238/223044935-97ab8d9f-8695-4a99-9b4d-7fecfaefeca6.png)

> the creator name changes to black bold on mobile, it should remain blue normal weight

Fixed

![image](https://user-images.githubusercontent.com/22791238/223045299-969fd5ed-8389-4ce9-9ea0-5471a3bd049e.png)

roiLeo commented:
> And then we may delete buy now by default from collections? Thinking out loud. Wdyt @roiLeo

Sure, we should show all items on collection item page by default.

roiLeo commented on layouts/explore-layout.vue:
```diff
@@ -15,14 +19,14 @@
           </div>
         </section>
-        <hr class="m-0" />
+        <hr class="text-color" />
```
Now we have a whitespace if you remove this

![Screenshot 2023-03-06 at 09-13-40 Denver waifus 🤎 KodaDot - NFT Market Explorer](https://user-images.githubusercontent.com/9987732/223054062-d2df955b-c195-41ff-9b05-c893d22bf466.png)

daiagi replied:
yep yep, missed that.
Thank you

exezbcz commented:
yes, that was wrong on my end then :D 

last changes:
- the whitespace 
- remove buy now by default from collection 

thanks!

daiagi commented:
> yes, that was wrong on my end then :D
> 
> last changes:
> 
>     * the whitespace
> 
>     * remove buy now by default from collection
> 
> 
> thanks!

Done

roiLeo reviewed (commented):
Don't know if it's on mobile spec but I see stuff changed:
- creator link should be blue
![Screenshot 2023-03-06 at 11-38-42 Ancient Civilizations KodaDot - NFT Market Explorer](https://user-images.githubusercontent.com/9987732/223087344-8a2b2320-e41b-4e2b-aae2-c1139e050c7d.png)
- explore layout spacing issue (compare it with beta)
![Screenshot 2023-03-06 at 11-40-27 Low minting fees and carbonless NFTs KodaDot - NFT Market Explorer](https://user-images.githubusercontent.com/9987732/223087681-ac07dccf-79f5-4307-b753-48d46e292f91.png)
- weird animation on close sidebar filter

roiLeo commented on components/collection/CollectionInfo.vue:
```diff
@@ -0,0 +1,112 @@
+    <div class="is-flex is-flex-direction-column is-flex-grow-1 max-width">
+      <div v-if="collectionInfo?.currentOwner" class="is-flex mb-2">
+        <div class="mr-2">{{ $t('activity.creator') }}</div>
+        <nuxt-link :to="`/${urlPrefix}/u/${address}`">
+          <IdentityIndex
+            ref="identity"
+            :address="address"
+            link-class="has-text-link" />
+        </nuxt-link>
```
```suggestion
        <nuxt-link :to="`/${urlPrefix}/u/${address}`" class="has-text-link">
          <IdentityIndex
            ref="identity"
            :address="address" />
        </nuxt-link>
```

daiagi replied:
I'm afraid this alone is not enough
on mobile, and maybe other cases, i did not explore deeply, ```IdenitityIndex``` renders ```IdentityLink``` and the applied class on ```nuxt-link``` just doesn't work

I've solved it before using css in ```collectionInfo``` which you requested changed

so this time i pass a prop through

on desktop, or any other case where ```IdentityIndex``` renders ```IdenitityPopover``` your solution is required

roiLeo replied:
is it still a problem?

roiLeo commented on components/identity/module/IdentityChain.vue:
```diff
@@ -50,6 +51,7 @@ defineProps<{
   address?: Address
   shortenedAddress?: string
   name?: string | object
+  linkClass?: string
```
I would remove this prop

roiLeo reviewed (commented):
`collectionStatsByIdWithOffers` retrieve all data on client browser :/

daiagi commented:
> weird animation on close sidebar filter

exists on beta as well, should solve in separate issue

> explore layout spacing issue (compare it with beta)

indeed, apologies. fixed

roiLeo commented:
> if i remove the `has-text-link` class from the internal `<a>` tag the text is black

Can't reproduce it, we have same stuff going in `GalleryItemDescription` component so I want to unify how it looks

daiagi commented:
> > if i remove the `has-text-link` class from the internal `<a>` tag the text is black
> 
> Can't reproduce it, we have same stuff going in `GalleryItemDescription` component so I want to unify how it looks

toggle mobile device in dev tools and make sure it us "mobile" mode aka  touch enabled. refresh page

![image](https://user-images.githubusercontent.com/22791238/223127209-9c956b9a-041e-478b-b461-57a74ec9ed09.png)

from ```IdentityLink.vue```
![image](https://user-images.githubusercontent.com/22791238/223127468-c986ede6-0a20-408a-88f0-abd31b608444.png)


maybe remove this ```!isMobileDevice```  => render identityPopover on touch devices as well ? wtyt @roiLeo

roiLeo commented:
> toggle mobile device in dev tools and make sure it is in "mobile" mode aka touch enabled. refresh page

Aight, I can reproduce it with mobile mode.

> from `IdentityLink.vue` 

This is where issue come from, since `IdentityLink` provide a nested `<a/>` tag, IMO we should replace this component by something else since when you change `a` tag to `div` it work like a charm

> maybe remove this `!isMobileDevice` => render identityPopover on touch devices as well ? wtyt @roiLeo

yep easy solution, but I think this has been introduce for a particular reason

daiagi commented:
Mind approving this PR and I'll open a new issue to deal with this?

roiLeo reviewed (approved):
What need to be improved:
- profile link on mobile (`IdentityLink` component)
- use `NeoSidebar` instead of Buefy Sidebar (`MobileFilter` component)
- `index` folder in `/components/explore` ???
- remove `linkClass` prop
- remove `class="mb-5"` (total in `ItemsGrid` & `Gallery` component)
- change `position: relative` with `is-relative` helper
- `v-if` on `ActiveCount` usage

otherwise lgtm

daiagi commented:
> What need to be improved:
> - profile link on mobile (`IdentityLink` component)
> - use `NeoSidebar` instead of Buefy Sidebar (`MobileFilter` component)
> - `index` folder in `/components/explore` ???
> - remove `linkClass` prop
> - remove `class="mb-5"` (total in `ItemsGrid` & `Gallery` component)
> - change `position: relative` with `is-relative` helper
> - `v-if` on `ActiveCount` usage

All done except:

#### Leftovers

- [ ] profile link on mobile (IdentityLink component)
- [ ] remove linkClass prop

daiagi commented:
@Jarsen136 @preschian  could i have your review please?

Jarsen136 reviewed (approved):
code lgtm
